src/renderer/components/cluster-manager/clusters-menu.tsx

```diff
@@ -112,26 +115,33 @@ export class ClustersMenu extends React.Component<Props> {
                   ref={provided.innerRef}
                   {...provided.droppableProps}
                 >
-                  {clusters.map((cluster, index) => (
-                    <Draggable draggableId={cluster.id} index={index} key={cluster.id}>
-                      {(provided: DraggableProvided) => (
-                        <div
-                          ref={provided.innerRef}
-                          {...provided.draggableProps}
-                          {...provided.dragHandleProps}
-                        >
-                          <ClusterIcon
-                            key={cluster.id}
-                            showErrors={true}
-                            cluster={cluster}
-                            isActive={cluster.id === getMatchedClusterId()}
-                            onClick={() => this.showCluster(cluster.id)}
-                            onContextMenu={() => this.showContextMenu(cluster)}
-                          />
-                        </div>
-                      )}
-                    </Draggable>
-                  ))}
+                  {clusters.map((cluster, index) => {
+                    const matched = matchPath(navigation.location.pathname, {
+                      path: [clusterViewRoute.path, clusterSettingsRoute.path].flat(),
+                      exact: true
+                    })
+                    const isActive = activeClusterId === cluster.id && !!matched;
```

Comment:
Why do we need to check matching the route to highlight icon?
Why `const isActive = cluster.id === clusterStore.activeClusterId` not enough?

Reply:
Btw you don't even check `matched.params.clusterId` in your code :)
Should be like `cluster.id === matched?.params.clusterId` in your case..

Reply:
Made suggested changes and completely removed `matchPath`.